src/util/autofilereloader.cpp

```diff
@@ -0,0 +1,23 @@
+#include "util/autofilereloader.h"
+
+#include <QDebug>
+#include <QString>
+
+#include "moc_autofilereloader.cpp"
+#include "util/runtimeloggingcategory.h"
+
+AutoFileReloader::AutoFileReloader(const RuntimeLoggingCategory& loggingCategory)
+        : m_loggingCategory(loggingCategory) {
+    connect(&m_fileWatcher,
+            &QFileSystemWatcher::fileChanged,
+            this,
+            &AutoFileReloader::slotFileChanged);
+}
+
+void AutoFileReloader::slotFileChanged(const QString& changedFile) {
+    qCDebug(m_loggingCategory) << "File" << changedFile << "has been changed.";
+    if (m_pathTriggeringReload.remove(changedFile)) {
```

Comment:
Why is a changed file removed form the watcher?

Reply:
See the comment in the header file. It is done to prevent double-reload caused by the different editor saving strategies (VSCode vs. vim/sed).

@Swiftb0y I think it should be pointed out in a comment that it's the callers responsibility to re-add the file once the reload was handled (in the QML case, this is done in the intercept call, after the changed file was parsed).

Reply:
Its in the header of `AutoFileReloader` as documentation for the `m_pathTriggeringReload` member.